snap/quota/quota.go

```diff
@@ -832,6 +852,40 @@ func (grp *Group) NewSubGroup(name string, resourceLimits Resources) (*Group, er
 	return subGrp, nil
 }
 
+// VerifyNesting takes a group and verifies that it satisfies the following conditions:
+// 1. That if any parent is mixed (has both snaps and sub-groups), it must be the immediate
+//    parent group.
+// 2. If the group itself is mixed, that it has only one level of sub-grouping.
+func (grp *Group) VerifyNesting() error {
```

Comment:
as this is about mixing snaps and nesting, maybe the name needs to be more specific?